### ISSUE TITLE
SQL Server driver creating ErrorException

### ISSUE THREAD

**Issue by ivan-au** [dg/dibi#398]
Version: 4.2.2

### Bug Description
When one of the parameters given to `\Dibi\Connection` are incorrect (e.g. incorrect password) error PHP `trying to access array offset on value of type null on Dibi\Drivers\SqlsrvDriver.php`  is generated.

### Steps To Reproduce
The bug can be reproduced when connecting to an SQL Server using the `sqlsrv` driver.
```
try {
	$db = new \Dibi\Connection([
		'driver'   => 'sqlsrv',
		host'     => 'host',
		'username' => 'user',
		'password' => 'wrong-password',  # deliberately changed to reproduce the error
		'database' => 'database',
		'profiler' => [
		    'file' => true,
		],
    ];
} catch (\Dibi\DriverException $e) {
	echo get_class($e) . ' ' . $e->getMessage());
}
```

### Expected Behavior
A `\Dibi\DriverException` should be generated.

### Possible Solution
Looking at the code at `Dibi\Drivers\SqlsrvDriver.php` I believe the issue is due to statement `sqlsrv_configure('WarningsReturnAsErrors', 1);` in line 68, which clears the SQL Server error; hence when invoking `$info = sqlsrv_errors(SQLSRV_ERR_ERRORS)` in line 72 the value of `$info` is null; resulting in the PHP error I describe.

**Comment by dg** [comment]
Can you check 4.2-dev, ie this commit https://github.com/dg/dibi/commit/0b394a993d274f15c0e30e1d9d05e27709a7be0f ?

**Comment by ivan-au** [comment]
Thanks @dg , that worked. Are you planning on merging to main branch?

**Comment by ivan-au** [comment]
> Thanks @dg , that worked. Are you planning on merging to main branch?

Just noticed it's been merged to the main branch, but it's not in the latest release 4.2.2 so it's not being picked up by Composer. Will you generate a new release any time soon?

**Comment by dg** [comment]
It would just be good if you could test it, and then I would release a new version. 

```
composer require dibi/dibi:v4.2.x-dev
```

**Comment by ivan-au** [comment]
Worked like a charm @dg 👍. I can see the DEV version in my `composer.json`. 

```
    "require": {
        "dibi/dibi": "v4.2.x-dev"
    }
```

I am now getting the expected `Dibi\DriverException` instead of the PHP error

**Comment by ivan-au** [comment]
FYI @dg , this issue was resolved in release 4.2.3. I'm closing the issue now.
